Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct AddCustomDynamicEntityDataSourceView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The point on the screen the user tapped.
+    @State private var tappedScreenPoint: CGPoint?
+    
+    /// The placement of the callout.
+    @State private var calloutPlacement: CalloutPlacement?
+    
+    /// A dynamic entity observation that was selected on the screen.
+    @State private var selectedObservation: DynamicEntityObservation?
```

Comment:
This doesn't actually need to be stored. Just get it from the callout placement itself:

```swift
.callout(placement: $calloutPlacement.animation(.default.speed(2))) { placement in
    let attributes = (placement.geoElement as! DynamicEntityObservation).attributes
    ...
}
```